src/lib.rs

```diff
@@ -80,14 +80,29 @@ struct Reporter {
 #[pymethods]
 impl Reporter {
     #[new]
-    fn new() -> PyResult<Reporter> {
+    fn new(config: Option<&PyDict>) -> PyResult<Reporter> {
+        let mut agent_host_name: String = "127.0.0.1".to_string();
+        let mut agent_port: i32 = 6831;
+
+        if let Some(config) = config {
+            agent_host_name = config
+                .get_item("agent_host_name")
+                .unwrap()
+                .extract()
+                .unwrap_or("127.0.0.1".to_string());
+            agent_port = config
+                .get_item("agent_port")
+                .unwrap()
+                .extract()
+                .unwrap_or(6831);
+        }
```

Comment:
Is there a better idiomatic Rust pattern to do this?

Reply:
Yes, see https://github.com/erikjohnston/rust-jaeger-python-client/pull/8#discussion_r954739087